### PR TITLE
Add classes

### PR DESCRIPTION
**Pull request by diegoyon**
### Add Person, Student, and Teacher classes

- Created class Person with specified variables and methods.
- Created class Student with specified variables and methods.
- Created class Teacher with specified variables and methods.
- Each class is saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    @age = age
+    @name = name
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    return true if of_age?
+
+    false
+  end
```
Comment (ShadyShawkat) [inline person.rb]: - [x] Kindly make sure that this method returns `true` if the `of_age?` returns `true`, _**or if the person has parent permission**_, as per [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=or%20if%20they%20have%20permission%20from%20parents.) requirement.

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(age, classroom, name = 'Unknown')
+    super(name, age)
```
Comment (ShadyShawkat) [inline student.rb]: - [x] Please add `parent_permission` to the constructor parameters of the `Student` class and pass it down to the `Person` class using `super()`.

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(age, specialization, name = 'Unknown')
+    super(name, age)
```
Comment (ShadyShawkat) [inline teacher.rb]: - [x] Please add `parent_permission` to the constructor parameters of the `Teacher` class and pass it down to the `Person` class using `super()`.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @diegoyon, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by m05e5 — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @diegoyon   🔥 

Good job on implementing the requested changes. You smashed it. Well Done.
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

goodjob


Now you can merge this branch 🚀 

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

_m05e5_

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**